contracts/ServiceNodeContribution.sol

```diff
@@ -143,6 +141,8 @@ contract ServiceNodeContribution is Shared {
             require(contributorAddresses.length > 0, "Operator has not contributed funds");
         }
 
+        maxContributors = stakingRewardsContract.maxContributors();
+        stakingRequirement = stakingRewardsContract.stakingRequirement();
```

Comment:
If the staking requirement is revised in a way where the following code passes for a pre-existing contract there's a chance that the minimum contribution will be incorrect for the contributors and the operator (e.g. if staking requirement goes up). 

All the code in this contract is written with the assumption that the operator is correctly minimally collateralised and allowing the staking requirement to change opens up a series of edge cases that we should probably avoid.

My recommendation is to avoid this entirely and make this change instead just

```
require(stakingRequirement == stakingRewardsContract.stakingRequirement(), "The staking requirement has changed and this contract can no longer be used for registering a Session Node. Please cancel the contract and start up a new one to continue");
```

There's a similar problem with the number of contributors changing. I'd have a similar error message for that.